quantum/send_string/send_string.c

```diff
@@ -208,19 +211,30 @@ void send_char(char ascii_code) {
 
     if (is_shifted) {
         register_code(KC_LEFT_SHIFT);
+        wait_ms(interval);
     }
+
     if (is_altgred) {
         register_code(KC_RIGHT_ALT);
+        wait_ms(interval);
     }
+
     tap_code(keycode);
```

Comment:
@fauxpark  Would `tap_code_delay` be better here?
```suggestion
    tap_code_delay(keycode, interval);
```

Reply:
Yeah that makes a lot of sense; done.

Reply:
No, that controls the time between keydown and keyup of the keycode, not the time between keyup of the previous event and keydown of the next.

Reply:
Hm, I was thinking it would make sense to use a consistent delay for every key event. Even without using the same delay for tapping, the input delay is what was used between key down for the mods and the ASCII character. If we want to distinguish, then should we use `TAP_CODE_DELAY` between key down on mods and the ASCII character, then the input delay after key ups? This seems like it could get confusing for users trying to figure out the right timing for `SEND_STRING _DELAY`.

In my opinion, it makes more sense to use a consistent delay for every key event. But I'll defer to your judgement.

Reply:
> In my opinion, it makes more sense to use a consistent delay for every key event.

This would be my expectation too.

Reply:
Then we should rename `TAP_CODE_DELAY` to more effectively communicate this. Right now it kind of implies that it's the delay between keydown and keyup when you call `tap_code()`, because it is.

Reply:
> Then we should rename `TAP_CODE_DELAY` to more effectively communicate this. Right now it kind of implies that it's the delay between keydown and keyup when you call `tap_code()`, because it is.

I mean I'm not too fussed about just using the tap code delay either way, so if this is a point of contention, I'm perfectly happy to just continue with previous semantics.

But with that said, I'm having trouble understanding what you're saying. In my view, `TAP_CODE_DELAY` is "delay between keydown and keyup when you call `tap_code()`", as you say. But the delay at issue here is the one in use in `SEND_STRING_DELAY`, and is, from my perspective, independent of `tap_code()`. The fact that it calls `tap_code()` internally is just an implementation detail. So to rename a long-standing configuration parameter because of its use as the default in an unrelated function seems puzzling to me.

Reply:
> But with that said, I'm having trouble understanding what you're saying.

You're suggesting we use `TAP_CODE_DELAY` as the default value for *every* delay that sendstring does, which includes the ones you're introducing, between the key events. Not just in the `tap_code()`. To me this muddies the meaning of the define's name, so if we want this value to be applied to more than just `tap_code()`, it should probably be renamed to be more generic. I'm not saying that should be done in *this* PR, just noting a potential point of confusion for users.